### ISSUE TITLE
Failure to compile

### ISSUE THREAD

**Issue by dmolik** [heketi/heketi#1261]
### Kind of issue

Bug

### Observed behavior

```
go build -i -ldflags "-X main.HEKETI_VERSION=v7.0.0-43-g9c40a0e -extldflags '-z relro -z now'" -o heketi
# github.com/heketi/heketi/pkg/kubernetes
pkg/kubernetes/backupdb.go:30:32: cannot use c (type *"github.com/heketi/heketi/vendor/k8s.io/client-go/rest".Config) as type *"github.com/heketi/heketi/vendor/k8s.io/kubernetes/vendor/k8s.io/client-go/rest".Config in argument to clientset.NewForConfig
pkg/kubernetes/pv.go:35:44: cannot use "github.com/heketi/heketi/vendor/k8s.io/apimachinery/pkg/api/resource".MustParse(fmt.Sprintf("%vGi", volume.VolumeInfo.VolumeCreateRequest.Size)) (type "github.com/heketi/heketi/vendor/k8s.io/apimachinery/pkg/api/resource".Quantity) as type "github.com/heketi/heketi/vendor/k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/api/resource".Quantity in assignment
make: *** [Makefile:62: heketi] Error 2
```

### Expected/desired behavior

make / make all doesn't eject an error?

### Details on how to reproduce (minimal and precise)

```
go get -u github.com/heketi/heketi
cd $GOPATH/github.com/heketi/heketi
make
```
### Information about the environment:

- Heketi version used: `master`
- Operating system used:  
  - ```
      uname -a
      Linux titan02 4.17.4-gentoo #1 SMP Tue Jul 3 12:49:17 EDT 2018 x86_64 AMD FX(tm)-8350 Eight-Core Processor AuthenticAMD GNU/Linux
      ```
   - ```
      go version
      go version go1.10.3 linux/amd64
      ```
  - ```
     gcc -v
     gcc version 7.3.0 (Gentoo Hardened 7.3.0-r3 p1.4) 
    ```
- Heketi compiled from sources
- no container
- direct install


### Other useful information

**Comment by phlogistonjohn** [comment]
Hi, that error looks like an issue of compatilibiyt between heketi's library dependencies.

How did you install the dependencies, via glide? Did you do anything other than using the rules in the makefile? And, what is the version of glide in use?

**Comment by dmolik** [comment]
glide version 0.13.1
make glide failed (glide install -v)
```
[INFO]	Replacing existing vendor dependencies
[INFO]	Removing nested vendor and Godeps/_workspace directories...
[INFO]	Removing: /home/dan/code/go/src/github.com/heketi/heketi/vendor/github.com/docker/distribution/vendor
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x30 pc=0x6fd642]
```
so I ran glide install without the -v

**Comment by dmolik** [comment]
okay so as it turns out I installed glide manually, and this seems to work now

**Comment by phlogistonjohn** [comment]
Hmm. That could be related (-v for glide install does not mean verbose) so we should try to resolve the glide failure and try again. IIRC building glide with go1.10 can trigger an issue that looks very similar. Could you try re-running `glide install -v` but using a version of glide build with go prior to 1.10? You ought to be able to use the packages from [glide's releases](https://github.com/Masterminds/glide/releases) if you're willing to do so.

**Comment by phlogistonjohn** [comment]
> okay so as it turns out I installed glide manually, and this seems to work now

Good to hear! If the entire build process works can we close this issue?

**Comment by dmolik** [comment]
so I tried it out with one of the bin versions from Masterminds, and it worked as well.

**Comment by dmolik** [comment]
Given that this is an issue with go1.10 and should be fixed with the 0.13.2 release of glide, I guess I'm okay with closing this issue

**Comment by dmolik** [comment]
On the flip side I should probably just use heketi-deps
